### PR TITLE
add openmp for clang backend

### PR DESCRIPTION
**Pull request by cloud11665**
I will post llama timing benchmarks soon.

### CODE REVIEW

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/cstyle.py                 4               0
tinygrad/runtime/ops_clang.py              4               4
------------------------------------------------------------
total                                      8               4
------------------------------------------------------------
lines added in the tinygrad folder: 4
```

**Comment by cloud11665** [comment]
image
llama

**Comment by cloud11665** [comment]
image
I've added parallelization of all global+local loops as there were cases where we were missing out on performance due to n_cores > loop_idx

File: tinygrad/codegen/cstyle.py [hunk 1]
```diff
@@ -177,6 +181,7 @@ def kk(s): kernel.append("  "*depth+s)
     [") {\n"] + list(prekernel) + ['\n'.join(kernel), "\n}"])
 
   if lang.half_prekernel and any(x.dtype == dtypes.float16 for x in bufs): prg = ''.join([f"{lang.half_prekernel}", "\n", prg])
+  prg = prg.replace("FIRST_LOOP_PRAGMA_REPLACEME", lang.loop_pragma(nloop) if lang.loop_pragma and not any(isinstance(x, LocalBuffer) for x in bufs) else "")
```
Comment (geohot) [inline tinygrad/codegen/cstyle.py:184]: Hmm, this looks tacky. Does it have to be done this way?

**Comment by geohot** [comment]
I won't merge with replaceme. There's a lot more flexibility now with CStyleLanguage

**Comment by geohot** [comment]
Going to close, can't merge `FIRST_LOOP_PRAGMA_REPLACEME`

**Pull request closed without merging**